agents/chopper_mc2000b/chopper_mc2000b_agent.py

```diff
@@ -0,0 +1,339 @@
+        the chopper blades for the controller to measure and set frequency.
+
+        Parameters
+        __________
+        reference : str
+            Reference mode of the blade. Default set to 'internalinner'.
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_blade_reference') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            reference = params['reference']
+
+            if reference in ('external', 'internal'):
+                ref = reference_mode_keys[reference]
+            else:
+                ref = reference_high_prec_mode[reference]
+
+            MC2000BSetReference(self.hdl, ref)
+
+        return True, "Chopper blade reference set to {}".format(params['reference'])
+
+    def acq(self, session, params):
+        """acq()
+
+        **Process** - Acquire data from the MC2000B chopper device.
+
+        """
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start acq because {self.lock.job} "
+                              "is already running")
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            last_release = time.time()
+
+            self.take_data = True
+
+            self.log.info("Starting data acquisition for {}".format(self.agent.agent_address))
+
+            while self.take_data:
+                # Relinquish sampling lock occasionally
+                if time.time() - last_release > 1.:
+                    last_release = time.time()
+                    if not self.lock.release_and_acquire(timeout=10):
+                        self.log.warn(f"Failed to re-acquire sampling lock, "
+                                      f"currently held by {self.lock.job}.")
+                        continue
+
+                freq_in = [0]
+                MC2000BGetFrequency(self.hdl, freq_in)
+                input_freq = freq_in[0]
+
+                freq_out = [0]
+                MC2000BGetReferenceOutFrequency(self.hdl, freq_out)
+                output_freq = freq_out[0]
+
+                # Publish data
+                input_data = {'block_name': 'input_freqs',
+                              'timestamp': time.time(),
+                              'data': {'input_freqs': input_freq}
+                              }
+                output_data = {'block_name': 'output_freqs',
+                               'timestamp': time.time(),
+                               'data': {'output_freqs': output_freq}
+                               }
+
+                self.agent.publish_to_feed('input_freqs', input_data)
+                self.agent.publish_to_feed('output_freqs', output_data)
```

Comment:
We don't actually need two feeds here. Publishing to a single feed with different blocks/field names should be sufficient.

My suggestion would be to drop one of the feeds, and rename the remaining one something like 'chopper_freqs'.

Reply:
I was going to ask you about this and completely forgot; thanks!